simson/common/data_transformations.py

```diff
@@ -69,68 +70,30 @@ def gdp_regression(historic_stocks_pc, gdppc, prediction_out, fitting_function_t
     assert len(shape_out) == 3, "Prediction array must have 3 dimensions: Time, Region, Good"
     pure_prediction = np.zeros_like(prediction_out)
     n_historic = historic_stocks_pc.shape[0]
-    i_lh = n_historic - 1
 
     for i_region in range(shape_out[1]):
         for i_good in range(shape_out[2]):
-
-            def sigmoid_fitting_function(prms):
-                return (
-                    prms[0] / (1. + np.exp(prms[1]/gdppc[:n_historic,i_region]))
-                    ) - historic_stocks_pc[:,i_region,i_good]
-                # Lagrangian multiplier to ensure matching last point:
-                # + prms[2] * prms[0] / (1. + np.exp(prms[1]/gdppc[i_lh,0])) - stocks_pc[-1,0,i] )
-
-            def exponential_fitting_function(prms):
-                return (
-                    prms[0] * (1 - np.exp(-prms[1]*gdppc[:n_historic,i_region]))
-                ) - historic_stocks_pc[:,i_region,i_good]
-
+            regional_historic_good = historic_stocks_pc[:, i_region, i_good]
+            regional_gdppc = gdppc[:, i_region]
             if fitting_function_type == 'sigmoid':
-                prms_out = least_squares(
-                    sigmoid_fitting_function,
-                    x0=np.array([2.*gdppc[i_lh,i_region], historic_stocks_pc[-1,i_region,i_good]]),
-                    gtol=1.e-12
-                )
-
-                pure_prediction[:, i_region, i_good] = prms_out.x[0] / (
-                        1. + np.exp(prms_out.x[1] / gdppc[:, i_region])
+                extrapolation = SigmoidalExtrapolation(
+                    data_to_extrapolate=regional_historic_good,
+                    extrapolate_from=regional_gdppc
```

Comment:
This is nitpicking, but the condition could only choose the function, and the call with arguments could be outside of it.

Reply:
Haha, yes, I also thought about this! For now, I'm fine with it as is, but definitely if we add more options with the same parameters I would go for that.